examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import List
+
+from langchain.schema import format_document
+from langchain_core.documents import Document
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.prompts.prompt import PromptTemplate
+from langchain_core.pydantic_v1 import BaseModel
+from langchain_core.runnables import (
+    ConfigurableField,
+    RunnableParallel,
+)
+from langchain_core.runnables.utils import ConfigurableFieldSingleOption
+from langchain_openai import ChatOpenAI
+
+from zep_python import ZepClient
+from zep_python.langchain import ZepVectorStore
+
+ZEP_API_KEY = os.environ.get("ZEP_API_KEY", None)  # Required for Zep Cloud
+
+ZEP_COLLECTION_NAME = os.environ.get("ZEP_COLLECTION", "langchaintest")
+
+if ZEP_API_KEY is None:
+    raise ValueError(
+        "ZEP_API_KEY is required for Zep Cloud. "
+        "Remove this check if using Zep Open Source."
+    )
+
+zep = ZepClient(
+    api_key=ZEP_API_KEY,
+)
+
+# Initialize ZepVectorStore
+vectorstore = ZepVectorStore(
+    collection_name=ZEP_COLLECTION_NAME,
+    zep_client=zep,
+)
+
+# Zep offers native, hardware-accelerated MMR. Enabling this will improve
+# the diversity of results, but may also reduce relevance. You can tune
+# the lambda parameter to control the tradeoff between relevance and diversity.
+# Enabling is a good default.
+retriever = vectorstore.as_retriever().configurable_fields(
+    search_type=ConfigurableFieldSingleOption(
+        id="search_type",
+        options={"Similarity": "similarity", "Similarity with MMR Reranking": "mmr"},
+        default="Similarity with MMR Reranking",
+        name="Search Type",
+        description="Type of search to perform: 'similarity' or 'mmr'",
+    ),
+    search_kwargs=ConfigurableField(
+        id="search_kwargs",
+        name="Search kwargs",
+        description=(
+            "Specify 'k' for number of results to return and 'lambda_mult' for tuning"
+            " MMR relevance vs diversity."
+        ),
+    ),
+)
+
+# RAG answer synthesis prompt
+template = """Answer the question based only on the following context:
+    <context>
+    {context}
+    </context>"""
+answer_prompt = ChatPromptTemplate.from_messages(
+    [
+        ("system", template),
+        ("user", "{question}"),
+    ]
+)
+
+DEFAULT_DOCUMENT_PROMPT = PromptTemplate.from_template(template="{page_content}")
+
+
+# User input
+class UserInput(BaseModel):
+    question: str
+
+
+def _combine_documents(
+    docs: List[Document],
+    document_prompt: PromptTemplate = DEFAULT_DOCUMENT_PROMPT,
+    document_separator: str = "\n\n",
+):
+    doc_strings = [format_document(doc, document_prompt) for doc in docs]
+    return document_separator.join(doc_strings)
+
+
+_inputs = RunnableParallel(
+    {"question": lambda x: x["question"], "context": retriever | _combine_documents},
```

Comment:
Where is the standalone question being generated? There's no runnable with a call to `question_synthesis`?

Reply:
So we're not generating a question as this example doesn't have message history. We extract the question field as part of input for ChatPromptTemplate here.